packages/parser/src/types/index.ts

```diff
@@ -0,0 +1,3 @@
+export type { ParserOptions } from './ParserOptions.js';
+export type { Envelope } from './envelope.js';
+export * from './schema.js';
```

Comment:
Besides the `*` (see comment under `packages/parser/src/schemas/index.ts`), why are we exporting the schemas here?

These are Zod objects so it's better if we don't mix them with pure TS types. One benefit of this is that in the future when we put this in the Lambda layer we can cut off anything under `packages/parser/src/types` and make the layer smaller and more performant.

As a rule of thumb, if we're not doing `export type ...`, it shouldn't be under `pacakges/*/src/types` but either be under the default export or its own subpath.

Reply:
Probably the wording is misleading, these are not schema types, but types that we infer from the built-in schema, that can use during development (#1838). If customers bring `@types/aws-lambda` there might be a mismatch between development and runtime. 

```ts
import type { ALBEvent } from "@aws-lambda-powertools/parser/types";

@parser({ schema: AlbSchema })
public async handler(event: ALBEvent, _context: Context): Promise<unknown> {
  return event;
}
```

Reply:
Ok my bad here.

I saw `./schema.js` and assumed it was `packages/parser/src/schema` rather than `packages/parser/src/types/schema.ts`.

Makes sense, I'd just remove the `*` then but the import is fine.